applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1139,12 +1188,13 @@ static void disconnect(const void *subscriber_id, uint8_t report_id)
 
 	if (rd->linked_rs == rs) {
 		clear_report_data(rd);
-		rd->linked_rs = find_next_report_state(rd);
+		rd->linked_rs = find_next_report_state(rd, rs);
+		rd->linked_rs->linked_rd = rd;
```

Comment:
`rd->linked_rs` may be NULL if nothing is connected... then the dereference fails.
move line 1192 under the condition (line 1196)